.github/scripts/periodic_merge.py

```diff
@@ -0,0 +1,146 @@
+
+PRS_WITH_LABEL_QUERY = (GH_PULL_REQUEST_FRAGMENT
+                        + GH_PR_REVIEWS_FRAGMENT
+                        + GH_CHECKSUITES_FRAGMENT
+                        + GH_COMMIT_AUTHORS_FRAGMENT) + """
+query ($owner: String!, $name: String!, $labels: [String!], $with_labels: Boolean = false) {
+  repository(owner: $owner, name: $name) {
+    pullRequests(first: 10, labels: $labels, states: OPEN){
+        nodes{
+        ...PullRequestFragment
+      }
+    }
+  }
+}
+"""
+
+
+def fetch_land_pending_pr_numbers(org: str, project: str) -> Any:
+    pr_query = gh_graphql(PRS_WITH_LABEL_QUERY,
+                          owner=org,
+                          name=project,
+                          labels=[LAND_PENDING_LABEL, LAND_QUEUED_LABEL],
+                          with_labels=True)
+    return pr_query['data']['repository']['pullRequests']['nodes']
+
+
+def get_github_prs(org: str, project: str, pr_query: Any) -> List[GitHubPR]:
+    prs = []
+    for pr_info in pr_query:
+        prs.append(GitHubPR(org, project, pr_info['number'], pr_info))
+    return prs
+
+
+def is_land_check(labels: List[str]) -> bool:
+    return LAND_CHECK_LABEL in labels
+
+
+def is_all_green(labels: List[str]) -> bool:
+    return ALL_GREEN_LABEL in labels
+
+def has_label(labels: List[str], label: str) -> bool:
+    return ALL_GREEN_LABEL in labels
+
+def validate_all_green(pr: GitHubPR) -> None:
+    pending = pr_get_pending_checks(pr)
+    failing = pr_get_failed_checks(pr)
+    if len(failing) > 0:
+        raise RuntimeError(f"{len(failing)} additional jobs have failed, first few of them are: " +
+                           ' ,'.join(f"[{x[0]}]({x[1]})" for x in failing[:5]))
+    if len(pending) > 0:
+        raise MandatoryChecksMissingError(f"Still waiting for {len(pending)} additional jobs to finish, " +
+                                          f"first few of them are: {' ,'.join(x[0] for x in pending[:5])}")
+
+def update_label_state(org: str, project: str, pr_num: int, old_state: str, new_state: str) -> None:
+    try:
+        gh_remove_label(org, project, pr_num, old_state)
+        gh_add_labels(org, project, pr_num, [new_state])
+    except Exception:
+        return
+
+
+def get_pr_link(org: str, project: str, pr_num: int) -> str:
+    return f"[PR {pr_num}](https://github.com/{org}/{project}/pull/{str(pr_num)})"
+
+def main() -> None:
+    # Find all the PRs that have land-pending label
+    repo = GitRepo(get_git_repo_dir(), get_git_remote_name())
+    org, project = repo.gh_owner_and_name()
+    pr_query = fetch_land_pending_pr_numbers(org, project)
+    prs = get_github_prs(org, project, pr_query)
+
+    pending_prs = ", ".join([str(pr.pr_num) for pr in prs])
+    print(f"Attempting to merge PRs: {pending_prs}" if len(prs) > 0
+          else "No land-pending pull requests found. Attempting to merge later")
+    check_for_sev(org, project, False)
+
+    # Validate each PR and merge them in
+    for pr in prs:
```

Comment:
Just curious, roughly how long does it take to process all pending PR on average? I remember you mention a figure like less than a minute or so

Reply:
Fetching all the PRs takes a bit since it's making an API call but the rest of this doesn't make any calls so around 30s-1m in total. The only time it takes longer than that is when it needs to merge in a PR.